djstripe/webhooks.py

```diff
@@ -34,26 +38,65 @@
 
 def handler(event_types):
     """
-    Decorator which registers a function as a webhook handler for the given
-    types of webhook events
+    Decorator which registers a function as a webhook handler for the specified
+    event types (e.g. 'customer') or fully qualified event sub-types (e.g.
+    'customer.subscription.deleted').
+
+    If an event type is specified then the handler will receive callbacks for
+    ALL webhook events of that type.  For example, if 'customer' is specified
+    then the handler will receive events for 'customer.subscription.created',
+    'customer.subscription.updated', etc.
+
+    :param event_types: The event type(s) or sub-type(s) that should be handled.
+    :type event_types: A sequence (`list`) or string (`str`/`unicode`).
     """
-    def decorator(f):
+    if isinstance(event_types, six.string_types):
+        event_types = [event_types]
+
+    def decorator(func):
         for event_type in event_types:
-            registrations[event_type].append(f)
-        return f
+            registrations[event_type].append(func)
+        return func
 
     return decorator
 
 
-def handler_all(f):
+def handler_all(func=None):
     """
     Decorator which registers a function as a webhook handler for ALL webhook
-    events
+    events, regardless of event type or sub-type.
     """
-    registrations_global.append(f)
-    return f
+    if not func:
+        return functools.partial(handler_all)
+
+    registrations_global.append(func)
+    return func
 
 
 def call_handlers(event, event_data, event_type, event_subtype):
-    for handler_func in registrations_global + registrations[event_type]:
+    """
+    Invokes all handlers for the provided event type/sub-type.
+
+    The handlers are invoked in the following order:
+
+        1. Global handlers
+        2. Event type handlers
+        3. Event sub-type handlers
+
+    Handlers within each group are invoked in order of registration.
+
+    :param event: The event model object.
+    :type event: `djstripe.models.Event`
+    :param event_data: The raw data for the event.
+    :type event_data: `dict`
+    :param event_type: The event type, e.g. 'customer'.
+    :type event_type: string (`str`/`unicode`)
+    :param event_subtype: The event sub-type, e.g. 'updated'.
+    :type event_subtype: string (`str`/`unicode`)
+    """
+    qualified_event_type = "%s.%s" % (event_type, event_subtype)
```

Comment:
Please use the new python string format:

``qualified_event_type = "{event_type}.{event_subtype}".format(event_type=event_type, event_subtype=event_subtype)``